fdbserver/include/fdbserver/IPager.h

```diff
@@ -362,6 +325,27 @@ class ArenaPage : public ReferenceCounted<ArenaPage>, public FastAllocated<Arena
 			}
 		}
 	};
+
+	struct AESEncryptionV1EncodingHeader {
```

Comment:
Am not sure if I understand the need for this structure; it seems to contains `EncryptionHeader`, but, then exposes `encode` and `decode` functionality; that seems bit odd. 

Encode/decode should happen at very few code instances (ideally one), it would better just to inline the code at the call-site. 
If not, then maybe rename it to `AESEncryptHelper` or something similar (`Header` also doing encryption/decryption feels odd to me)
Thanks

Reply:
I too feel it a bit odd. Does renaming it to XXXEncoder sound right to you?